src/hdx_cli/cli_interface/dictionary/commands.py

```diff
@@ -46,23 +47,28 @@ def files(ctx: click.Context):
                'usercontext': profileinfo}
 
 
-@click.command(help='Create dictionary.')
-@click.argument('dictionary_description_file')
+@click.command(help=f"Create dictionary. 'dictionary_file' contains the settings. "
+                    "The filename and name in settings will be replaced by 'dictionary_filename' "
+                    "and 'dictionary_name' respectively.")
+@click.argument('dictionary_file')
+@click.argument('dictionary_filename')
 @click.argument('dictionary_name')
 @click.pass_context
 @report_error_and_exit(exctype=Exception)
 def create_dict(ctx: click.Context,
-                dictionary_description_file,
+                dictionary_file: str,
+                dictionary_filename: str,
                 dictionary_name: str):
     user_profile = ctx.parent.obj['usercontext']
     resource_path = ctx.parent.obj['resource_path']
-    # FIXME: basic_create does not work because it does local opening
-    # of a file. A dictionary file is remote.
-    basic_create(user_profile,
-                 resource_path,
-                 dictionary_name,
-                 dictionary_description_file)
-    print(f'Created dictionary {dictionary_name}.')
+    with open(dictionary_file, 'r', encoding='utf-8') as input_body:
+        dictionary_body = json.load(input_body)
+        dictionary_body['settings']['filename'] = dictionary_filename
+    basic_create_with_body_from_string(user_profile,
+                                       resource_path,
+                                       dictionary_name,
+                                       json.dumps(dictionary_body))
```

Comment:
So this dictionary file isn't the actual dictionary file. It's a different one that just has the dictionary settings in it? I really don't think that should be in a file. Either way, if it is or isn't, we need to validate the settings and put some error handling here. If there are any problems in that file, this code will error.

Reply:
Regarding this, I think we need that the user specifies a settings file of the dictionary because it has several requires fields to be created (layout, output_columns, primary_key, format).
On the other hand, errors in the file format, or if fields are missing from the settings file, the CLI displays the error returned by the API.
Anyway, I'm going to add a validation before this -> `dictionary_body['settings']['filename'] = dictionary_filename`
Because it can break.
Thank you!